pkg/controller/machine/machine_controller_test.go

```diff
@@ -17,87 +17,217 @@ limitations under the License.
-	if err := c.Create(context.TODO(), instance); err != nil {
-		t.Fatalf("error creating instance: %v", err)
+		By("Creating the Machine")
+		Expect(k8sClient.Create(ctx, instance)).To(Succeed())
+
+		By("Setting the AuthoritativeAPI to ClusterAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityClusterAPI
+		})).Should(Succeed())
+
+		By("Verifying that the AuthoritativeAPI is set to Cluster API")
+		Eventually(func() (machinev1.MachineAuthority, error) {
+			if err := k8sClient.Get(ctx, objectKey(instance), instance); err != nil {
+				return "", err
+			}
+			return instance.Status.AuthoritativeAPI, nil
+		}, timeout).Should(Equal(machinev1.MachineAuthorityClusterAPI))
+
+		By("Verifying the paused condition is approproately set to true")
+		Eventually(func() ([]machinev1.Condition, error) {
+			if err := k8sClient.Get(ctx, objectKey(instance), instance); err != nil {
+				return nil, err
+			}
+
+			// The condition is set to true
+			return instance.Status.Conditions, nil
+		}, timeout).Should(ContainElement(SatisfyAll(
+			HaveField("Type", Equal(PausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionTrue)),
+		)))
+
+		By("Transitioning the AuthoritativeAPI though 'Migrating' to MachineAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityMigrating
+		})).Should(Succeed())
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityMachineAPI
+		})).Should(Succeed())
+
+		By("Verifying the paused condition is approproately set to false")
+		Eventually(func() ([]machinev1.Condition, error) {
+			if err := k8sClient.Get(ctx, objectKey(instance), instance); err != nil {
+				return nil, err
+			}
+
+			// The condition is set to false
+			return instance.Status.Conditions, nil
+		}, timeout).Should(ContainElement(SatisfyAll(
+			HaveField("Type", Equal(PausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionFalse)),
+		)))
+	})
+})
+
+func objectKey(object metav1.Object) client.ObjectKey {
+	return client.ObjectKey{
+		Namespace: object.GetNamespace(),
+		Name:      object.GetName(),
 	}
-	defer func() {
-		if err := c.Delete(context.TODO(), instance); err != nil {
-			t.Fatalf("error deleting instance: %v", err)
-		}
-	}()
-	g := NewWithT(t)
-	g.Eventually(func() (machinev1.MachineStatus, error) {
-		machine := &machinev1.Machine{}
-		namespacedName := client.ObjectKey{Namespace: instance.Namespace, Name: instance.Name}
-		err := c.Get(ctx, namespacedName, machine)
-		if err != nil {
-			return machinev1.MachineStatus{}, err
+}
+
+func cleanResources() error {
+	machines := &machinev1.MachineList{}
+	if err := k8sClient.List(ctx, machines); err != nil {
+		return err
+	}
+	for _, machine := range machines.Items {
+		m := machine
+		if err := k8sClient.Delete(ctx, &m); err != nil {
```

Comment:
I believe there's a DeleteAll call you can do

Or try the cleanup utils on the actuator package side

Reply:
Bump, still to look at